jda4/src/main/java/me/scarsz/jdaappender/ChannelLoggingHandler.java

```diff
@@ -236,25 +241,47 @@ private Message updateMessage() throws IllegalStateException {
         // safeguard against empty lines
         while (full.contains("\n\n")) full = full.replace("\n\n", "\n");
 
-        if (currentMessage != null) {
-            try {
-                return currentMessage.editMessage(full).submit().get();
-            } catch (Exception e) {
-                if (this.isInterruptedException(e)) return currentMessage;
-                Throwable cause = e.getCause();
-                if (cause instanceof ErrorResponseException
-                        && ((ErrorResponseException) cause).getErrorResponse() == ErrorResponse.UNKNOWN_MESSAGE) {
-                    currentMessage = null;
-                } else {
-                    throw new RuntimeException(e.getCause());
+        try {
+            return sendOrEditMessage(full, channel);
+        } catch (ErrorResponseException errorResponseException) {
+            if (errorResponseException.getErrorResponse() == ErrorResponse.UNKNOWN_MESSAGE) {
+                currentMessage = null;
+                // in case the message was deleted and the original message had a harmful link, catch the error again
+                try {
+                    return sendOrEditMessage(full, channel);
+                } catch (ErrorResponseException errorResponseException2) {
+                    if (errorResponseException2.getErrorCode() == MESSAGE_BLOCKED_BY_HARMFUL_LINK_FILTER_ERROR_CODE) {
+                        full = URL_PATTERN.matcher(full).replaceAll("$1");
+                        return sendOrEditMessage(full, channel);
+                    }
+                    throw new RuntimeException(errorResponseException2);
                 }
+            } else if (errorResponseException.getErrorCode() == MESSAGE_BLOCKED_BY_HARMFUL_LINK_FILTER_ERROR_CODE) {
+                full = URL_PATTERN.matcher(full).replaceAll("$1");
+                return sendOrEditMessage(full, channel);
+            } else {
+                throw new RuntimeException(errorResponseException);
```

Comment:
Alternative retry that reduces a bit of code duplication (untested), thoughts?

```suggestion
        try {
            for (int i = 0; i < 2; i++) {
                try {
                    return sendOrEditMessage(full, channel);
                } catch (ErrorResponseException ex) {
                    if (i == 0 && getErrorResponse() == ErrorResponse.UNKNOWN_MESSAGE) {
                        currentMessage = null;
                        continue;
                    }
                    throw ex;
                }
            }
        } catch (ErrorResponseException ex) {
            if (ex.getErrorCode() == MESSAGE_BLOCKED_BY_HARMFUL_LINK_FILTER_ERROR_CODE) {
                full = URL_PATTERN.matcher(full).replaceAll("$1");
                return sendOrEditMessage(full, channel);
            } else {
                throw new RuntimeException(errorResponseException);
```
also is that last wrap in RuntimeException wanted?

Reply:
code suggestion LGTM. question answered in discord.